clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/loadbalancer/UpsertAmazonLoadBalancerAtomicOperation.groovy

```diff
@@ -110,9 +128,25 @@ class UpsertAmazonLoadBalancerAtomicOperation implements AtomicOperation<UpsertA
       } catch (AmazonServiceException ignore) {
       }
 
-      def securityGroups = regionScopedProvider.securityGroupService.
-              getSecurityGroupIds(description.securityGroups, description.vpcId).values()
+      def securityGroupNamesToIds = regionScopedProvider.securityGroupService.getSecurityGroupIds(description.securityGroups, description.vpcId)
+      if (description.name == description.application) {
+        try {
+          String applicationLoadBalancerSecurityGroupId = ingressApplicationLoadBalancerGroup(
+            description,
+            region,
+            listeners,
+            securityGroupLookupFactory
+          )
+
+          securityGroupNamesToIds.put(description.application + "-elb", applicationLoadBalancerSecurityGroupId)
+          task.updateStatus BASE_PHASE, "Autorized application Load Balancer Security Group $applicationLoadBalancerSecurityGroupId"
+        } catch (Exception e) {
+          log.error("Failed to complete authorizing app elb security group {}-elb on application security group {}", description.name, description.name, e)
+          task.updateStatus BASE_PHASE, "Failed to complete authorizing app elb security group $description.name-elb on application security group"
```

Comment:
Maybe `Failed to authorize app ELB security group ${description.name}-elb on application security group` instead?